### DISCUSSION TITLE
transform: translate()

### DISCUSSION THREAD

**Discussion by xDarkDelux** [mikke89/RmlUi#359] [Q&A]
hello, I have a task to place a document with an offset from the center. For example, you need to make an element offset (relative to the center of the context, not the top left edge) +200dp on the left and +90dp on the top, and you need to take into account that when the context is resized, these offsets are preserved. I do not know how to do it correctly, I found information that this can be done using margin: auto auto; and transform:translate(200dp, 25dp), and it would seem that this is my ideal solution, but .. inside the window I have a list, at first everything is fine, but when I started adding data to the list and when the list overflowed, a scroll appeared on the verticals when the appearance of the scroll - all the data of my list is gone. Randomly, I found that if the list has the attribute overflow:hidden auto; - then when a scroll appears - my data is shifted to their positions without taking into account transform:translate (while the window itself remains in its place with an offsettransform:translate), if you remove the overflow attribute - everything works fine, but then the overflow is not hidden and the scroll does not appear . How to be in this situation? Maybe there are other options for positioning documents relative to the center and not the top left edge? Is it a bug problem with transform:translate and overflow?

**Comment by mikke89** [comment]
Hm, it's possible that there are some issues with transform and clipping. If you can make a reproducible I can take a look at it.

With that said, I wouldn't really recommend using transform for layouting. What you could do instead is add an absolutely positioned element directly inside your body. So your body is centered using margins, but the inner element is used for the offset with e,g, top/left properties.

**Comment by xDarkDelux** [comment]
to reproduce the situation:
in this example, when you have 8 lines in the list - your text disappears, when you have 7 lines in the list - everything is fine
(don't forget to change font-family)
> <rml>
    <head>
        <link type="text/rcss" href="rml.rcss"/>
        <title></title>
        <style>
            body
            {
                width: 168dp;
                height: 170dp;
                margin: auto auto;
                transform: translate(200dp, 25dp);
                font-family: Tahoma;
            }
            tbody td
            {
                height: 20dp;
                white-space: nowrap;
                overflow: hidden;
                font-size: 11dp;
                line-height: 17dp;
            }
        </style>
    </head>
    <body id="TestWnd">
        <div style="width: 152dp; height: 166dp;">
            <div id="TestList" style="display: block; width: 146dp; height: 140dp; margin-left: 3dp; margin-top: 4dp;">
                <div style="width: 146dp; height: 140dp; overflow:hidden auto;">
                    <table>
                        <col id="col_0" style="width: 146dp;"/>
                        <tbody>
                            <tr><td style="padding-left: 5dp;"> text 1 </td></tr>
                            <tr><td style="padding-left: 5dp;"> text 2 </td></tr>
                            <tr><td style="padding-left: 5dp;"> text 3 </td></tr>
                            <tr><td style="padding-left: 5dp;"> text 4 </td></tr>
                            <tr><td style="padding-left: 5dp;"> text 5 </td></tr>
                            <tr><td style="padding-left: 5dp;"> text 6 </td></tr>
                            <tr><td style="padding-left: 5dp;"> text 7 </td></tr>
                            <tr><td style="padding-left: 5dp;"> text 8 </td></tr>
                        </tbody>
                    </table>
                </div>
            </div>
        </div>
    </body>
</rml>

**Reply by mikke89** [reply]
Thanks. Yeah, I see that the text clipping doesn't take the current transform into account. I'll see if we can come up with some solution to that, or we might just disable it whenever transforms are present.
